code/WorkInProgress/Cael_Aislinn/Rust/core_field.dm

```diff
@@ -208,9 +222,57 @@ Deuterium-tritium fusion: 4.5 x 10^7 K
 			dormant_reactant_quantities[reactant] -= radiate
 			radiation += radiate
 
+	transfer_energy()
+
+	for(var/mob/living/l in range(src, round((radiation / 50000) ** 0.25 + (size / 2))))
+		var/rads = (radiation / 1000) * sqrt(1/(max(get_dist(l, src), 1)))
+		l.apply_effect(rads, IRRADIATE)
+
+	var/instability = round(((temp - warning_point) / (meltdown_point - warning_point)) * 100)
+	if(temp > warning_point) // while the core is still damaged and it's still worth noting its status
+
+		if((world.timeofday - lastwarning) / 10 >= warning_delay)
+			var/warning = ""
+			var/offset = 0
+
+			// Damage still low.
+			if(temp >= temp_archived) // The damage is still going up
+				warning = "Danger! EM Field instability detected, now at [instability]%."
+				offset=150
+
+				if(temp > danger_point)
+					if(owned_core.emagged)
+						warning = "EM FIELD INSTABILITY AT [instability]%. Safety overrides will %//SZZZZZZZHT-instability."
+					else
+						warning = "EM FIELD INSTABILITY AT [instability]%. Safety overrides will activate at 100% instability."
+					offset=0
+					//audio_offset = 100
+			if(temp < temp_archived)
+				warning = "EM Field returning to safe operating levels. Instability: [instability]%"
+			//radio.say(warning, "Supermatter [short_name] Monitor")
+			var/datum/speech/speech = radio.create_speech(warning, frequency=1459, transmitter=radio)
+			speech.name = "R-UST Monitor"
+			speech.job = "Automated Announcement"
+			speech.as_name = "R-UST Monitor"
+			Broadcast_Message(speech, level = list(0,1))
+			returnToPool(speech)
+
+			lastwarning = world.timeofday - offset
+			temp_archived = temp
+
+		if (temp > meltdown_point && !owned_core.emagged)
+			Destroy()
```

Comment:
Okay but seriously though why the fuck do you think you use Destroy() manually to do this, I need to know. Where in the world are people getting this idea?